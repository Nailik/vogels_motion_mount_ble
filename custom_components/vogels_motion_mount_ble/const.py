"""Constants for the Vogels Motion Mount BLE integration."""

DOMAIN = "vogels_motion_mount_ble"

SERVICE_UUID = "3e6fe65d-ed78-11e4-895e-00026fd5c52c"

CHAR_NAME_UUID = "c005fa37-0651-4800-b000-000000000000"
CHAR_SIZE_UUID = "c005fa2b-0651-4800-b000-000000000000"
CHAR_DISTANCE_UUID = "c005fa00-0651-4800-b000-000000000000"
CHAR_ROTATION_UUID = "c005fa01-0651-4800-b000-000000000000"
CHAR_AUTOMATIC_UUID = "c005fa02-0651-4800-b000-000000000000"
CHAR_FREEZE_UUID = "c005fa14-0651-4800-b000-000000000000"
CHAR_PRESET_UUIDS = [
    "c005fa0a-0651-4800-b000-000000000000",
    "c005fa0b-0651-4800-b000-000000000000",
    "c005fa0c-0651-4800-b000-000000000000",
    "c005fa0d-0651-4800-b000-000000000000",
    "c005fa0e-0651-4800-b000-000000000000",
    "c005fa0f-0651-4800-b000-000000000000",
    "c005fa10-0651-4800-b000-000000000000",
]
CHAR_PRESET_UUID = "c005fa2a-0651-4800-b000-000000000000"
