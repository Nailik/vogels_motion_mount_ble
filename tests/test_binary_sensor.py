import pytest
from syrupy.assertion import SnapshotAssertion
from homeassistant.const import Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er
from unittest.mock import patch

from . import setup_integration

from pytest_homeassistant_custom_component.common import (
    MockConfigEntry,
    snapshot_platform,
)


@pytest.mark.usefixtures("enable_bluetooth", "patch_default_bleak_client")
async def test_all_entities(
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
    mock_config_entry: MockConfigEntry,
    entity_registry: er.EntityRegistry,
) -> None:
    """Test all entities."""
    with patch(
        "custom_components.vogels_motion_mount_ble.PLATFORMS", [Platform.NUMBER]
    ):
        await setup_integration(hass, mock_config_entry)

    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)
