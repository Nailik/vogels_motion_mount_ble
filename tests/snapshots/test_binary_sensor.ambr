# serializer version: 1
# name: test_all_entities[binary_sensor.mount_connection-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.mount_connection',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.CONNECTIVITY: 'connectivity'>,
    'original_icon': 'mdi:wifi',
    'original_name': 'Connection',
    'platform': 'vogels_motion_mount_ble',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'connection',
    'unique_id': 'connection',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[binary_sensor.mount_connection-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'connectivity',
      'friendly_name': 'Mount Connection',
      'icon': 'mdi:wifi',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.mount_connection',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
